### PR TITLE
Add support to Apple silicon packages

### PR DESCRIPTION
**Pull request by DFolchA**
|Related issue|
|---|
|#2086|

## Description

This PR adds support to compile and generate Apple silicon packages using the tools provided in the wazuh-packages repository.

### CODE REVIEW

**Review by jnasselle — CHANGES_REQUESTED** [review]

File: macos/generate_wazuh_packages.sh [hunk 1]
```diff
@@ -150,14 +150,16 @@ function build_package() {
...
+        echo "The wazuh agent package for MacOS has been successfully built."
+        pkg_name="wazuh-agent-${VERSION}-${REVISION}.${ARCH}.pkg"
         sign_pkg
         notarize_pkg
         if [[ "${CHECKSUM}" == "yes" ]]; then
             mkdir -p ${CHECKSUMDIR}
             cd ${DESTINATION} && shasum -a512 "${pkg_name}" > "${CHECKSUMDIR}/${pkg_name}.sha512"
         fi
+        ls -l ${DESTINATION}/${pkg_name}
+        ls -l ${DESTINATION}/wazuh-agent-${VERSION}-${REVISION}*
```
[the remark this answers is not in the document]
Comment (vikman90) [reply macos/generate_wazuh_packages.sh:162]: It seems that they are for debugging purposes.

**Review by jnasselle — APPROVED** [review]

File: macos/generate_wazuh_packages.sh [hunk 2]
```diff
@@ -236,6 +239,10 @@ function install_deps() {
         echo "Something went wrong installing packagesbuild."
     fi
 
+    if [ "$(uname -m)" = "arm64" ]; then 
+        echo "Installing dependencies for arm64 architecture"
```
[the remark this answers is not in the document]
Comment (jnasselle) [reply macos/generate_wazuh_packages.sh]: Solved by f0603f12f982241e200ab76cb417c3eaa35e72d4

File: macos/generate_wazuh_packages.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jnasselle) [reply macos/generate_wazuh_packages.sh]: Solved by 435f532eab82ee88641ab4f58679052dedf24ef0

File: macos/generate_wazuh_packages.sh [hunk 3]
```diff
@@ -150,14 +150,16 @@ function build_package() {
 
     # create package
     if packagesbuild ${AGENT_PKG_FILE} --build-folder ${DESTINATION} ; then
-        echo "The wazuh agent package for MacOS X has been successfully built."
-        pkg_name="wazuh-agent-${VERSION}-${REVISION}.pkg"
+        echo "The wazuh agent package for MacOS has been successfully built."
```
[the remark this answers is not in the document]
Comment (jnasselle) [reply macos/generate_wazuh_packages.sh]: Solved by 24fda07b45fa91dfd687a49bc23711b56aa0a4a7

**Pull request merged**